### PR TITLE
Release 2.6

### PR DESCRIPTION
**Pull request by StorytellerCZ**
* Remove deprecated APIs
* Add support for arbitrary iterables in #each templates

### CODE REVIEW

**Comment by StorytellerCZ** [comment]
My apologies for the crazy commits, had some issues with Git due to setting once upstream branch wrong. Should be all fixed now.

**Review by filipenevola — CHANGES_REQUESTED** [review]
Include this new package in the publish.sh script

**Comment by jankapunkt** [comment]
@StorytellerCZ what's the status here / is there something todo left?

**Comment by StorytellerCZ** [comment]
I would like to include #358 into this as well as the final piece. Also waiting for approval from @filipenevola to proceed.

**Comment by fredmaiaarantes** [comment]
@StorytellerCZ @jankapunkt, I have updated the test-app to Meteor 2.7 and all the tests are ok.

From your point of view this PR is ready to be merged? If so, I can do it today.

**Comment by jankapunkt** [comment]
I can't review before Monday but then I would like to do so.

**Comment by fredmaiaarantes** [comment]
> I can't review before Monday but then I would like to do so. 

No problem, you can just ping me.

**Comment by StorytellerCZ** [comment]
Let's include #358 into this as well.

**Comment by jankapunkt** [comment]
Agree 👍

File: packages/observe-sequence/observe_sequence_tests.js [hunk 1]
```diff
@@ -205,6 +205,19 @@ Tinytest.add('observe-sequence - initial data for all sequence types', function
...
+    return new Map();
+  }, function () {}, []);
+
+  // non-array iterable (non-empty)
+  if(typeof Set == 'function') runOneObserveSequenceTestCase(test, function () {
+    return new Set([{foo: 1}, {bar: 2}]);
+  }, function () {}, [
+    {addedAt: [0, {foo: 1}, 0, null]},
+    {addedAt: [1, {bar: 2}, 1, null]}
+  ]);
```
Comment (jankapunkt) [inline packages/observe-sequence/observe_sequence_tests.js:220]: Should we add a few more tests here? I think if we publish this and productions will use this there may be unforseen edge-cases.

I would propose to add tests for:

- [x] empty map
- [ ] arbitrary filled map
- [ ] empty set
- [x] arbitrary filled set
- [ ] empty custom structure implementing the iterable protocol
- [ ] arbitrary filled custom structure implementing the iterable protocol

File: packages/spacebars/spacebars-runtime.js [hunk 2]
```diff
@@ -278,6 +278,3 @@ Spacebars.With = function (argFunc, contentFunc, elseFunc) {
 
   return view;
 };
-
-// XXX COMPAT WITH 0.9.0
-Spacebars.TemplateWith = Blaze._TemplateWith;
```
Comment (jankapunkt) [inline packages/spacebars/spacebars-runtime.js:283]: We should add a note to the changelog/history,  that this API is removed

File: packages/templating-runtime/templating.js [hunk 3]
```diff
@@ -26,18 +26,6 @@ Template.__checkName = function (name) {
...
-  Template.__checkName(name);
-  Template[name] = new Template("Template." + name, renderFunc);
-  // Exempt packages built pre-0.9.0 from warnings about using old
-  // helper syntax, because we can.  It's not very useful to get a
-  // warning about someone else's code (like a package on Atmosphere),
-  // and this should at least put a bit of a dent in number of warnings
-  // that come from packages that haven't been updated lately.
-  Template[name]._NOWARN_OLDSTYLE_HELPERS = true;
-};
-
```
Comment (jankapunkt) [inline packages/templating-runtime/templating.js:40]: We should add a note to the changelog/history,  that this API is removed

File: packages/templating-runtime/templating.js [hunk 4]
```diff
@@ -175,12 +163,3 @@ Template._migrateTemplate = function (templateName, newTemplate, migrate) {
...
 };
-
-// XXX COMPAT WITH 0.9.0
-UI.body = Template.body;
-
-// XXX COMPAT WITH 0.9.0
-// (<body> tags in packages built with 0.9.0)
-Template.__body__ = Template.body;
-Template.__body__.__contentParts = Template.body.contentViews;
-Template.__body__.__instantiate = Template.body.renderToDocument;
```
Comment (jankapunkt) [inline packages/templating-runtime/templating.js:186]: We should add a note to the changelog/history,  that this API is removed

**Review by jankapunkt — CHANGES_REQUESTED** [review]
I think we should do a last round and add a few more tests for the new iterable feature to prevent Blaze users become unintended testers and add notes to the breaking changes in README or Changelog.

File: packages/observe-sequence/observe_sequence_tests.js (hunk 1 above)
Comment (StorytellerCZ) [reply packages/observe-sequence/observe_sequence_tests.js:220]: More tests are always welcome.

**Comment by StorytellerCZ** [comment]
I'll add notice about the removed APIs to history.

**Comment by fredmaiaarantes** [comment]
As soon as the comments are addressed, we can merge this PR.

File: HISTORY.md [hunk 5]
```diff
@@ -1,3 +1,8 @@
+## v2.6.0, 2021-June-XX
```
[the remark this answers is not in the document]
Comment (StorytellerCZ) [reply HISTORY.md]: Done!

File: packages/blaze-html-templates/package.js [hunk 6]
```diff
@@ -10,12 +10,6 @@ Package.onUse(function(api) {
     // A library for reactive user interfaces
     'blaze@2.5.0',
 
-    // The following packages are basically empty shells that just exist to
-    // satisfy code checking for the existence of a package. Rest assured that
-    // they are not adding any bloat to your bundle.
-    'ui@1.0.13', // XXX COMPAT WITH PACKAGES BUILT FOR 0.9.0.
-    'spacebars@1.2.0', // XXX COMPAT WITH PACKAGES BUILT FOR 0.9.0
-
```
[the remark this answers is not in the document]
Comment (StorytellerCZ) [reply packages/blaze-html-templates/package.js:18]: added

**Review by jankapunkt — APPROVED** [review]
Generally approved, only a minor change could be applied, which I personally think could be kept due to fun reasons :-D

File: HISTORY.md [hunk 7]
```diff
@@ -1,3 +1,24 @@
+## v2.6.0, 2022-April-XX
+
+* [#330](https://github.com/meteor/blaze/pull/330) Removed deprecated APIs from before Meteor 1.0
+  * This is potentially breaking, especially for old packages and apps.
+  * `blaze-html-templates@2.0.0`
+    * Dependency on `ui` and `spacebars` package has been removed
+  * `spacebars@1.3.0`
+    * `Spacebars.TemplateWith` has beer removed, please use `Blaze._TemplateWith` if you need it.
```
[the remark this answers is not in the document]
Comment (StorytellerCZ) [reply HISTORY.md:8]: That is a good catch :laughing: :beers: Would be a nice experiment to leave it there to see who read the changelogs.

**Review by fredmaiaarantes — APPROVED** [review]

**Comment by denihs** [comment]
Guys, everything is now published. Awesome work!

**Comment by StorytellerCZ** [comment]
@denihs great work! Looks like the docs page still needs to be manually updated.
